account-history/unsettled-amounts/demo.js

```diff
@@ -0,0 +1,242 @@
+        })
+    }
+    /**
+     * Clear the Dropdown specified
+     * @param {*} identifier 
+     */
+    function resetDropdownOptions(identifier) {
+        var select = document.getElementById("idCbx" + identifier)
+        var sel_len = select.length
+        for (var i = 0; i < sel_len; i++) {
+            select.remove(0)
+        }
+    }
+
+    /**
+     * Parse key information out of response and format for display
+     * @param {*} response 
+     * @param {*} requestType 
+     * @param {*} requestUrl 
+     * @param {*} params 
+     * @returns 
+     */
+    function parseResponse(response, requestType, requestUrl, params) {
+        if (requestType === "Instruments") {
+            return "Instruments for which amounts are owed\n" + "Endpoint: \n\t" + requestUrl + "\nParameters: \n\t" + params + "\n"
+        } else if (response === null || response.CashFlows[0] === undefined) {
+            return "No Amounts Owed\n" + "Endpoint: \n\t" + requestUrl + "\nParameters: \n\t" + params + "\n"
+        }
+        var details = []
+        var dateRange = "Date Range is: \n\t"
+            + response.CashFlows[0].ValueDate + " => "
+            + response.CashFlows[response.CashFlows.length - 1].ValueDate
+        var total = response.Total + " " + response.Currency
+
+        response = response[requestType]
+        response.forEach(elem => { details.push(elem.ExchangeId || elem.Currency) })
+        total = "Total (estimated) amount owed is: \n\t" + total
+        if (requestType === "Currencies") {
+            details = "Amounts are owed in these Currencies: \n\t" + details.toString()
+        } else {
+            details = "Amounts are owed to these Exchanges: \n\t" + details.toString()
+        }
+        return "Endpoint: \n\t" + requestUrl + "\nParameters: \n\t" + params + "\n" + total + "\n" + dateRange + "\n" + details + "\n\n"
+
+    }
+    /**
+     * Get the unsettled amounts for a specific client, by currency or amount types
+     * @return {void}
+     */
+    function getUnsettledAmountsByCurrency(scope) {
+
+        let parameters = "?ClientKey=" + encodeURIComponent(demo.user.clientKey) + "&Scope=" + scope
+        AmountTypeSource = read("AmountTypeSource")
+        if (AmountTypeSource !== "All") parameters += "&AmountTypeSource=" + AmountTypeSource
+        fetch(
+            demo.apiUrl + "/hist/v1/unsettledamounts" + parameters,
+            {
+                "method": "GET",
+                "headers": {
+                    "Authorization": "Bearer " + document.getElementById("idBearerToken").value
+                }
+            }
+        ).then(function (response) {
+            if (response.ok) {
+                response.json().then(function (responseJson) {
+                    if (scope === "AmountTypes" && responseJson.Currencies.length > 0) {
+                        resetDropdownOptions("CurrencyAndAmountType")
+                        responseJson.Currencies.forEach(elem => {
```

Comment:
elem should be 'currency'

Reply:
Why not:
responseJson.Currencies.forEach(currency => {

Reply:
missed this apparently? thought i resolve it :)

Reply:
fixed this part now